app/controllers/spree/api/retailops/settlement_controller.rb

```diff
@@ -62,14 +62,30 @@ def mark_complete
           render text: @settlement_results.to_json
         end
 
+        # This gets called by RetailOps after a returned item has been
+        # received and accepted at the warehouse. We need to look elsehwere
+        # for hook when returned item has been received but can not be restocked.
+        #
+        # We first look for a `RetailopsReimbursementHandler` class to see if the host
+        # application has defined any custom logic. If they have, we simply
+        # pass the parameters to the defined `Reimbursement.from_retailops_add_refund`
+        # factory method and call `RetailopsReimbursementHandler#create_reimbursement`.
         def add_refund
-          ActiveRecord::Base.transaction do
-            find_order
-            assert_return params
-            @order.update!
+          if defined?(RetailopsReimbursementHandler)
+            reimbursement = RetailopsReimbursementHandler.from_retailops_add_refund(
+              params['settlement']
```

Comment:
I forgot to include documentation in the README. The `RetailopsReimbursementHandler` is intended to be defined by the host application if, and only if, they would like to customize the behavior here. If it is not defined, the default behavior is used. If you were referring to `params['settlement']`, that is defined by the parameters sent to the controller by RetailOps.